### PR TITLE
CU-9p0j1t - Create account for private key

### PR DESCRIPTION
**Pull request by haieng1234**
Create account for private key done

### CODE REVIEW

**Comment by jamescwk** [comment]
Task linked: [CU-9p0j1t Create account for private key](https://app.clickup.com/t/9p0j1t)

File: src/modules/account/views/ViewAccountCreatePrivateKey.vue [hunk 1]
```diff
@@ -55,34 +62,39 @@ export default {
...
+          err.value = "Password for wallet " + walletState.currentLoggedInWallet.name + " is invalid" ;
+        } else {    
           // create account
-          const account = appStore.createNewAccountPrivateKey(appStore.state.currentLoggedInWallet.name, walletPassword.value, privKey.value, appStore);
-          // update to state
-          appStore.updateAccountState(account, accountName.value);
-          router.push({ name: "createdAccount", params: {publicKey: account.publicKey, privateKey: account.privateKey, address: appStore.pretty(account.address.address), name: accountName.value }});
-        }
-      }else{
+          const pubkey = Account.createFromPrivateKey(privKey.value,ChainUtils.getNetworkType(networkState.currentNetworkProfile.network.type));
```
Comment (sleepyOwl14) [inline src/modules/account/views/ViewAccountCreatePrivateKey.vue:74]: please use more accurate variable, it is account instance from the SDK

File: src/modules/account/views/ViewAccountCreatePrivateKey.vue [hunk 2]
```diff
@@ -55,34 +62,39 @@ export default {
...
+          const verifyExistingAccount = walletState.currentLoggedInWallet.accounts.find((element) => element.publicKey == pubkey.publicKey);
+          if (verifyExistingAccount) {
+            err.value = "Account with this Private Key is already in the wallet";
+          } else {          
+            let password = WalletUtils.createPassword(walletPassword.value);
+            const account = WalletUtils.generateNewAccount(ChainUtils.getNetworkType(networkState.currentNetworkProfile.network.type));
```
Comment (sleepyOwl14) [inline src/modules/account/views/ViewAccountCreatePrivateKey.vue:80]: why need to generateNewAccount ?

**Review by sleepyOwl14 — CHANGES_REQUESTED** [review]
please fix

**Review by sleepyOwl14 — APPROVED** [review]

**Pull request merged**